validation/markets-validator.js

```diff
@@ -0,0 +1,549 @@
+#!/usr/bin/env node
```

Comment:
For each function I need to know what the function does, what arguments it expects (meaning and type) and what it returns (meaning and type). E.g. (I'm guessing here)

````js
/*
 * Validator for multiple features
 *
 * Validates a set of features for a city.
 *
 * Constructor arguments:
 *   - `features`: a list of feature IDs
 *   - `cityName`: a city ID
 */
function FeaturesValidator(features, cityName) {
````

Ideally we should decide upon a documentation generator and use the appropriate doc syntax. We don't have to decide that now but if you're going to write docs anyway I see no point in delaying the decision. I don't have any experience with JS doc generators, but judging from [this Stack Overflow question](http://stackoverflow.com/questions/27334309/ecmascript-6-documentation-generators) it seems that [ESDoc](https://esdoc.org/) might be a good choice.